### ISSUE TITLE
drop Equals injection in DefineTypeConstraint

### ISSUE THREAD

**Issue by sanderr** [inmanta/inmanta-core#5787]
This used to be required but probably isn't anymore. inmanta/lsm#580 may be required so backwards compatibility should be taken into account.

```python
        if hasattr(expression, "arguments"):
            # some sort of function call
            expression = Equals(expression, Literal(True))
```

**Comment by arnaudsjs** [comment]
@sanderr What do you mean with the part about "backwards compatibility should be taken into account"? Do you mean that it should be verified that the removal of this if-block doesn't break anything?

**Comment by sanderr** [comment]
Yes. More specifically that we should consider whether there is any realistic scenario where a user would have the version of core with this removed while still having the version of the lsm module from before inmanta/lsm#580. I know that the systemtenant project currently has a `~=2.20.0` constraint but I think it is running iso6 and I expect they can move to a higher lsm version when they move to iso7. Nevertheless, we should probably check.

I do think we have some freedom if we only merge it into iso7 but since we have no way of making these sorts of inter-component dependencies explicit, I prefer to be careful about it anyways.

**Comment by sanderr** [comment]
(This fix was released in lsm 21.1.4)
